src/main/java/com/lpnu/shaggybeavers/service/impl/UserRegionServiceImpl.java

```diff
@@ -17,4 +19,10 @@ public class UserRegionServiceImpl extends CRUDServiceImpl<UserRegion,Long> impl
     protected JpaRepository<UserRegion, Long> getRepository () {
         return this.repository;
     }
+
+    @Override
+    public Optional<UserRegion> findByUserIdAndRegionId(Long userId, Long regionId) {
```

Comment:
you should return UserRegion because your logic builind over checking is exist and if you put OrElseThrow in that method you will be rid of from if else